docs/patterns/cache-aside.md

```diff
@@ -1,156 +1,135 @@
+
+```csharp
+private static ConnectionMultiplexer Connection;
+// We will default to a five minute expiration
+private const double DefaultExpirationTimeInMinutes = 5.0;
+
+public async Task<MyEntity> GetMyEntityAsync(int id)
+{
+  // Define a unique key for this method and its parameters.
+  var key = $"MyEntity:{id}";
+  var cache = Connection.GetDatabase();
+  
+  // Try to get the entity from the cache.
+  var json = await cache.StringGetAsync(key).ConfigureAwait(false);
+  var value = string.IsNullOrWhiteSpace(json) 
+                ? default(MyEntity) 
+                : JsonConvert.DeserializeObject<MyEntity>(json);
+  
+  if (value == null) // Cache miss
+  {
+    // If there's a cache miss, get the entity from the original store and cache it.
+    // Code has been omitted because it's data store dependent.  
+    value = ...;
+
+    // Avoid caching a null value.
+    if (value != null)
+    {
+      // Put the item in the cache with a custom expiration time that 
+      // depends on how critical it is to have stale data.
+      await cache.StringSetAsync(key, JsonConvert.SerializeObject(value)).ConfigureAwait(false);
+      await cache.KeyExpireAsync(key, TimeSpan.FromMinutes(DefaultExpirationTimeInMinutes)).ConfigureAwait(false);
+    }
+  }
+
+  return value;
+}
+```
+
+>  The examples use the Azure Redis Cache API to access the store and retrieve information from the cache. For more information, see [Using Microsoft Azure Redis Cache](https://docs.microsoft.com/en-us/azure/redis-cache/cache-dotnet-how-to-use-azure-redis-cache) and [How to create a Web App with Redis Cache](https://docs.microsoft.com/en-us/azure/redis-cache/cache-web-app-howto)
+
+The `UpdateEntityAsync` method shown below demonstrates how to invalidate an object in the cache when the value is changed by the application. This is an example of a write-through approach. The code updates the original data store and then removes the cached item from the cache by calling the `KeyDeleteAsync` method, specifying the key (the code has been omitted because it is data store dependent).
+
+>  The order of the steps in this sequence is important. If the item is removed before the cache is updated, the client application has a short period of time to fetch the data (because it isn't found in the cache) before the item in the data store has been changed, resulting in the cache containing stale data.
+
+```csharp
+public async Task UpdateEntityAsync(MyEntity entity)
+{
+    // Update the object in the original data store

```

Comment:
Comment should be followed by a line of code that shows the intent e.g.
```
await this.store.UpdateEntityAsync(entity).ConfigureAwait(false); 
```
or similar, otherwise readers may ignore the comments.